lmdeploy/serve/async_engine.py

```diff
@@ -31,67 +31,73 @@ def __init__(self, model_path, instance_num=32, tp=1, **kwargs) -> None:
                                                      tp=tp,
                                                      **kwargs)
         self.tokenizer = self.tm_model.tokenizer
-        self.generators = [
-            self.tm_model.create_instance() for i in range(instance_num)
-        ]
         self.instance_num = instance_num
         self.model = self.tm_model.model
-        self.available = [True] * instance_num
-        self.starts = [None] * instance_num
-        self.steps = {}
+        self.id2step = {}
+        self.id2generator = {}
         self.loop = asyncio.get_event_loop()
+        self.special_gen = self.tm_model.create_instance()
+        self.gens_set = set()
+        for i in range(instance_num):
+            self.gens_set.add(self.tm_model.create_instance())
 
     def stop_session(self, session_id: int):
         """Stop a session by a session_id."""
-        instance_id = session_id % self.instance_num
-        input_ids = self.tokenizer.encode('')
-        for outputs in self.generators[instance_id].stream_infer(
-                session_id,
-                input_ids,
-                request_output_len=0,
-                sequence_start=False,
-                sequence_end=False,
-                stop=True):
+        input_ids = [self.tm_model.eos_id]
```

Comment:
A model like `qwen` doesn't  have  `eos_id`. So does it work for qwen model?

Reply:
It should work since `eos_id` here is from tm_model. We converted `eod_id` of remote qwen model to `eos_id` in `Tokenizer` class.

Reply:
my mistake. It is `bos_id` that doesn't exist in qwen model